src/library/scala/collection/immutable/HashSet.scala

```diff
@@ -190,10 +190,10 @@ final class HashSet[A] private[immutable](private[immutable] val rootNode: Bitma
     * Stops iterating the first time that f returns `false`.*/
   @`inline` private[collection] def foreachWithHashWhile(f: (A, Int) => Boolean): Unit = rootNode.foreachWithHashWhile(f)
 
-  def subsetOf(that: Set[A]): Boolean = if (that.isEmpty) true else that match {
+  def subsetOf(that: Set[A]): Boolean = isEmpty || !that.isEmpty && (that match {
```

Comment:
LGTM.

I was wondering whether the fix should be here or in each of the receivers of `rootNode.subsetOf`. That initially felt preferable to also avoid this edge case when recursing into the trie. But I'm pretty sure `that.rootNode` is the only possible empty node in `that`.

There may be an optimization opportunity available here too, by short-circuiting when `this.size > that.size`. Or maybe that already falls out of the underlying implementation. The goal would be to avoid needless `.equals` calls.

Reply:
Thanks for :eyes:. On my exercise run just now, considered the case in the OP example, where common prefix is long. You could keep the prefix length at the node and avoid one-legged depth. Not sure if that will work out, but once bitten by the algo bug so to speak, one looks for more to do.